lib/std/crypto/bcrypt.zig

```diff
@@ -251,45 +386,106 @@ fn strHashInternal(password: []const u8, rounds_log: u6, salt: [salt_length]u8)
-    const salt_str = h[7..][0..salt_str_length];
-    var salt: [salt_length]u8 = undefined;
-    try Codec.decode(salt[0..], salt_str[0..]);
-    const rounds_log = fmt.parseInt(u6, rounds_log_str[0..], 10) catch return error.InvalidEncoding;
-    const wanted_s = try strHashInternal(password, rounds_log, salt);
-    if (!mem.eql(u8, wanted_s[0..], h[0..])) {
-        return error.PasswordVerificationFailed;
+pub fn strVerify(
+    str: []const u8,
+    password: []const u8,
+    _: VerifyOptions,
+) Error!void {
+    if (mem.startsWith(u8, str, crypt_format.prefix)) {
+        return CryptFormatHasher.verify(str, password);
+    } else {
+        return PhcFormatHasher.verify(str, password);
     }
 }
 
 test "bcrypt codec" {
     var salt: [salt_length]u8 = undefined;
     crypto.random.bytes(&salt);
     var salt_str: [salt_str_length]u8 = undefined;
-    Codec.encode(salt_str[0..], salt[0..]);
+    crypt_format.Codec.encode(salt_str[0..], salt[0..]);
     var salt2: [salt_length]u8 = undefined;
-    try Codec.decode(salt2[0..], salt_str[0..]);
+    try crypt_format.Codec.decode(salt2[0..], salt_str[0..]);
     try testing.expectEqualSlices(u8, salt[0..], salt2[0..]);
 }
 
-test "bcrypt" {
-    const s = try strHash("password", 5);
-    try strVerify(s, "password");
-    try testing.expectError(error.PasswordVerificationFailed, strVerify(s, "invalid password"));
-
-    const long_s = try strHash("password" ** 100, 5);
-    try strVerify(long_s, "password" ** 100);
-    try strVerify(long_s, "password" ** 101);
+test "bcrypt crypt format" {
+    const hash_options = HashOptions{
+        .allocator = null,
+        .params = .{ .rounds_log = 5 },
+        .encoding = .crypt,
+    };
+    const verify_options = VerifyOptions{ .allocator = null };
+
+    var buf: [hash_length]u8 = undefined;
+    const s = try strHash("password", hash_options, &buf);
+
+    try testing.expect(mem.startsWith(u8, s, crypt_format.prefix));
+    try strVerify(s, "password", verify_options);
+    try testing.expectError(
+        error.PasswordVerificationFailed,
+        strVerify(s, "invalid password", verify_options),
+    );
+
+    var long_buf: [hash_length]u8 = undefined;
+    const long_s = try strHash("password" ** 100, hash_options, &long_buf);
+
+    try testing.expect(mem.startsWith(u8, long_s, crypt_format.prefix));
+    try strVerify(long_s, "password" ** 100, verify_options);
+    try strVerify(long_s, "password" ** 101, verify_options);
+
+    try strVerify(
+        "$2b$08$WUQKyBCaKpziCwUXHiMVvu40dYVjkTxtWJlftl0PpjY2BxWSvFIEe",
+        "The devil himself",
+        verify_options,
+    );
+}
 
-    try strVerify("$2b$08$WUQKyBCaKpziCwUXHiMVvu40dYVjkTxtWJlftl0PpjY2BxWSvFIEe".*, "The devil himself");
+test "bcrypt phc format" {
+    const hash_options = HashOptions{
+        .allocator = null,
+        .params = .{ .rounds_log = 5 },
+        .encoding = .phc,
+    };
+    const verify_options = VerifyOptions{ .allocator = null };
```

Comment:
Is there a way to define `VerifyOptions` with a default value of `null` for `allocator` instead?

Reply:
It was for consistence)